net/src/headers.rs

```diff
@@ -319,6 +319,11 @@ impl Headers {
         Headers::default()
     }
 
+    /// Add / Replace Ethernet header
+    pub fn set_eth(&mut self, eth: Eth) {
```

Comment:
technically this method is unsafe in the sense that you can set an ethernet header with an ethertype which doesn't match the lower layers of the frame.  Will think about a good way to fix this.

Reply:
Having thought about it some more, I don't think we can fix this safety issue without requiring the user to go through a strongly typed builder API on every packet manipulation.

The actual fix here should be to make the API contract of the packet / headers types free form and then create a validated type wrapper which does not allow mutations without consuming the structure back into the mutable form.

That will prevent us from sending (or accepting) total nonsense while still giving us the latitude to actually manipulate packets.

That said, wherever possible, we should still forbid creating nonsense packets; moving errors closer to their actual cause point is just good practice anyway.

Reply:
I would say it's the caller's responsibility to set the ether-type. I mean, IMO, the net crate should provide the machinery to build packets, but can't be foreseeing all possible combinations and, if it ought to, it should probably be after the fact maybe when deparsing(), or via some validation method to be called explicitly?